src/main/java/com/commercetools/sync/categories/CategorySync.java

```diff
@@ -0,0 +1,194 @@
+
+import static com.commercetools.sync.commons.helpers.BaseSyncStatistics.getStatisticsAsJSONString;
+import static java.lang.String.format;
+
+public class CategorySync implements Sync {
+    private static final Logger LOGGER = LoggerFactory.getLogger(CategorySync.class);
+    private CategorySyncOptions syncOptions;
+    private CategorySyncStatistics statistics;
+
+    public CategorySync(@Nonnull final CategorySyncOptions syncOptions) {
+        this.syncOptions = syncOptions;
+    }
+
+    /**
+     * Traverses a {@link List} of {@link CategoryDraft} objects and tries to fetch a category, from the CTP project with
+     * the configuration stored in the {@code syncOptions} instance of this class, using the external id. If a category exists,
+     * this category is synced to be the same as the new category draft in this list. If no category exist with such external id,
+     * a new category, identical to this new category draft  is created.
+     * <p>
+     * The {@code statistics} instance is updated accordingly to whether the CTP request was carried out successfully or not.
+     * If an exception was thrown on executing the request to CTP,
+     * the optional error callback specified in the {@code syncOptions} is called.
+     *
+     * @param categoryDrafts the list of new category drafts to sync to the CTP project.
+     * @param <T>            the type of the resource draft. In this case CategoryDraft.
+     */
+    @Override
+    public <T> void syncDrafts(@Nonnull final List<T> categoryDrafts) {
+        statistics = new CategorySyncStatistics();
+        LOGGER.info(format("About to sync %d category drafts into CTP project with key '%s'."
+                , categoryDrafts.size(), getSyncOptions().getClientConfig().getProjectKey()));
+        for (int i = 0; i < categoryDrafts.size(); i++) {
+            try {
+                final CategoryDraft categoryDraft = (CategoryDraft) categoryDrafts.get(i);
+                if (categoryDraft != null) {
+                    getStatistics().incrementProcessed();
+                    final String externalId = categoryDraft.getExternalId();
+                    if (externalId != null) {
+                        final Category oldCategory = fetchCategory(externalId);
+                        if (oldCategory != null) {
+                            syncCategories(oldCategory, categoryDraft);
+                        } else {
+                            createCategory(categoryDraft);
+                        }
+                    } else {
+                        failSync(format("CategoryDraft with name: %s doesn't have an externalId.",
+                                categoryDraft.getName().toString()), null);
+                    }
+                }
+            } catch (ClassCastException e) {
+                failSync(format("Element at position %d is not an instance of CategoryDraft.", i), e);
+            }
+
+        }
+        getStatistics().calculateProcessingTime();
```

Comment:
It is calculated by subtracting finish from start. However, we don't need to show start and end times to end user. We only show the time it took to process in the final statistics of the sync, which look as follows:

````json
{
  "reportMessage": "Summary: 2 categories were processed in total (0 created, 0 updated and 0 categories failed to sync).",
  "updated": 0,
  "created": 0,
  "failed": 0,
  "processed": 2,
  "processingTimeInDays": 0,
  "processingTimeInHours": 0,
  "processingTimeInMinutes": 0,
  "processingTimeInSeconds": 3,
  "processingTimeInMillis": 3459,
  "humanReadableProcessingTime": "0d, 0h, 0m, 3s, 459ms"
}
````